src/main/java/com/es/phoneshop/controller/servlet/MiniCartServlet.java

```diff
@@ -0,0 +1,21 @@
+package com.es.phoneshop.controller.servlet;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+public class MiniCartServlet extends HttpServlet {
```

Comment:
You should consider either:
- Not taking any attributes directly from session within jsp files, in that case you need to attach cart attribute to request here (and also in CartPageServlet).
- Remove this class as it does nothing than including another servlet (minicart jsp one). In this case you might want to jsp:include it directly from master.tag / map "/cart/miniCart" directly to minicart.jsp in web.xml